### PR TITLE
test: add integration tests for router

### PR DESCRIPTION
**Pull request by fiam**
- chore: move demo subgraphs to individual Go packages
- fix: cleanup incorrect module references to the repo
- chore: allow stopping subgraphs
- test: add integration test for router
- test: add helper functions for integration test

## Motivation and Context

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router-tests/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+	zapLogger := zap.New(zapcore.NewCore(
+		zapcore.NewConsoleEncoder(ec),
+		syncer,
+		zapcore.ErrorLevel,
+	))
+	rs, err := core.NewRouter(
+		core.WithFederatedGraphName(cfg.Graph.Name),
+		core.WithStaticRouterConfig(routerConfig),
+		core.WithModulesConfig(cfg.Modules),
```
Comment (StarpTech) [inline router-tests/integration_test.go:163]: You don't need that if you have no modules.

File: router/cmd/custom/module/module_test.go [hunk 2]
```diff
@@ -3,12 +3,13 @@ package module
...
-	"github.com/stretchr/testify/assert"
-	"github.com/wundergraph/cosmo/router/core"
-	"github.com/wundergraph/cosmo/router/internal/config"
 	"net/http/httptest"
 	"os"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wundergraph/cosmo/router/config"
+	"github.com/wundergraph/cosmo/router/core"
```
Comment (StarpTech) [inline router/cmd/custom/module/module_test.go:12]: How can we now run this test in the CI with your integration test setup?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/cmd/router/main.go [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
 
 import (
+	"fmt"
+	"runtime"
+	"time"
+
 	routercmd "github.com/wundergraph/cosmo/router/cmd"
 )
 
+func printMemUsage() {
```
[the remark this answers is not in the document]
Comment (fiam) [reply router/cmd/router/main.go]: Fixed, thanks!

File: router-tests/integration_test.go (hunk 1 above)
Comment (fiam) [reply router-tests/integration_test.go:163]: Thanks! I've deleted it.

File: router-tests/runner_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	}
+	r.cmd.Wait()
+	return nil
+}
+
+func (r *subprocessSubgraphsRunner) Ports() []int {
+	return []int{4001, 4002, 4003, 4004}
+}
+
+func NewSubprocessSubgraphsRunner() (SubgraphsRunner, error) {
```
Comment (StarpTech) [inline router-tests/runner_test.go:81]: This won't run on Win correct?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router-tests/runner_test.go (hunk 4 above)
Comment (fiam) [reply router-tests/runner_test.go]: I've refactored it so it will fail with an appropriate error message on Windows

**Review by jensneuse — APPROVED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**